frontend/public/components/operator-lifecycle-manager/descriptors/status/index.tsx

```diff
@@ -79,7 +80,7 @@ export const StatusDescriptor: React.SFC<DescriptorProps> = (props) => {
     </div>
     <dd className="olm-descriptor__value">
       {descriptor.displayName === 'Status' ?
-        (<StatusIconAndText status={value} iconName={value === 'Running' ? 'ok' : undefined} />) :
+        ({value === 'Running' ? <SuccessStatus status={value} /> : <Status status={value} />) :
```

Comment:
Just some thoughts.

I'd like to be able to use the same `Status` component here instead. But we should then enumerate on a set of accepted status values in the prop types:
`<Status status={value === 'Running' && 'Complete' || value} />`

Alternatively, provide bool props (for the pre-defined finite set of concrete statuses) that take precedence over the status:
`<Status status={value} success={value === 'Running'} />`

Reply:
> Just some thoughts.
> 
> I'd like to be able to use the same `Status` component here instead. But we should then enumerate on a set of accepted status values in the prop types:
> `<Status status={value === 'Running' && 'Complete' || value} />`

I actually initially did this initially when I created this patch. But then I realised it is not possible as the status passed as prop is used as a `title` for the StatusIconAndText so the result in your example when `value === 'Running'` results in `Icon + 'Complete'` instead of `Icon + 'Running'`

> 
> Alternatively, provide bool props (for the pre-defined finite set of concrete statuses) that take precedence over the status:
> `<Status status={value} success={value === 'Running'} />`

I feel like expanding Status component to address this may be a bit overkill. What we need is to provide a way to define mapping of a certain (custom) status string to predefined status component (which ensures to render correct icon)

e.g.:
```
switch (status) {
  case 'discovered':
    return <SuccessStatus title={status} />;
  case 'discovering':
    return <ProgressStatus title={status} />;
  case 'discovery error':
    return <ErrorStatus title={status}>{errorMessage}</ErrorStatus>;
  default:
    return <Status status={status} /> // fall back to Status component for other states
}
```

This IMHO nicely says: "`discovered` is a `success` status"...

Comparing this to current solution which says: "Make 'Running' status act the same as 'Complete' status"

There is currently also one other way to define custom status existing implementation and that is by using `iconName` on the StatusIconAndText component. (see https://github.com/openshift/console/pull/1774/files#diff-2b92d7e552b962318adc922d3e6f4b61R39). I find this also quite bad as it forces you to specify icon name, which can lead to icon usage inconsistency and problems when icons are refactored/changed.

@spadgett noted a problem of ensuring to achieve consistency of same icon always being used for given status string. I think my changes don't make the current situation worse in this regard. Your example shows, there are cases when it is necessary to deviate from this. My changes add standardised status components which should help avoid this.

Reply:
heh of course i wasn't thinking about title being the status itself ... argh